include/ftxui/component/component.hpp

```diff
@@ -66,10 +66,20 @@ Component Renderer(std::function<Element()>);
 Component Renderer(std::function<Element(bool /* focused */)>);
 Component CatchEvent(Component child, std::function<bool(Event)>);
 Component Maybe(Component, const bool* show);
+Component Maybe(Component, std::function<bool()>&&);
 Component Collapsible(ConstStringRef label,
                       Component child,
                       Ref<bool> show = false);
 
+// -- Decorator --
+template<typename Decorator>
+Component operator|(Component component, Decorator&& decorator) {
+  return decorator(std::move(component));
+}
+using ComponentDecorator = std::function<Component(Component)>;
+ComponentDecorator CatchEvent(std::function<bool(Event)>&& on_event);
```

Comment:
@ArthurSonzogni can we use lower-case name for these decorators like what we have in `include/ftxui/dom/elements.hpp` ?  for two reasons

1. more consistent this way
2. avoid the name collisions -- so we can have a `renderer()` decorator.

Reply:
I am a bit sad about the different style being used, I agree. However using both `Renderer` and `renderer` would be even worse.
I don't believe there is any problem for the `renderer` component. I just added it.

Reply:
okay. commented inlined in the code.